### DISCUSSION TITLE
Error 500 - Too few arguments

### DISCUSSION THREAD

**Discussion by DevRaeph** [spatie/laravel-ray#168] [Q&A]
Got error 500 when calling "ray('Hello world');"

Log:

[2021-03-03 10:33:07] local.ERROR: Uncaught ArgumentCountError: Too few arguments to function Spatie\Ray\Origin\Origin::__construct(), 2 passed in E:\Development\gassner-webconnect\vendor\spatie\laravel-ray\src\OriginFactory.php on line 30 and exactly 3 expected in E:\Development\gassner-webconnect\vendor\spatie\ray\src\Origin\Origin.php:11
Stack trace:
#0 E:\Development\gassner-webconnect\vendor\spatie\laravel-ray\src\OriginFactory.php(30): Spatie\Ray\Origin\Origin->__construct()
#1 E:\Development\gassner-webconnect\vendor\spatie\ray\src\Payloads\Payload.php(56): Spatie\LaravelRay\OriginFactory->getOrigin()
#2 E:\Development\gassner-webconnect\vendor\spatie\ray\src\Payloads\Payload.php(42): Spatie\Ray\Payloads\Payload->getOrigin()
#3 E:\Development\gassner-webconnect\vendor\spatie\ray\src\Request.php(30): Spatie\Ray\Payloads\Payload->toArray()
#4 [internal function]: Spatie\Ray\Request->Spatie\Ray\{closure}()
#5 E:\Development\gassner-webconnect\vendor\spatie\ray\src\Request.php(31): array_map()
#6 E:\Development\gassner-webconnect\vendor\spa {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Uncaught ArgumentCountError: Too few arguments to function Spatie\\Ray\\Origin\\Origin::__construct(), 2 passed in E:\\Development\\gassner-webconnect\\vendor\\spatie\\laravel-ray\\src\\OriginFactory.php on line 30 and exactly 3 expected in E:\\Development\\gassner-webconnect\\vendor\\spatie\
ay\\src\\Origin\\Origin.php:11
Stack trace:
#0 E:\\Development\\gassner-webconnect\\vendor\\spatie\\laravel-ray\\src\\OriginFactory.php(30): Spatie\\Ray\\Origin\\Origin->__construct()
#1 E:\\Development\\gassner-webconnect\\vendor\\spatie\
ay\\src\\Payloads\\Payload.php(56): Spatie\\LaravelRay\\OriginFactory->getOrigin()
#2 E:\\Development\\gassner-webconnect\\vendor\\spatie\
ay\\src\\Payloads\\Payload.php(42): Spatie\\Ray\\Payloads\\Payload->getOrigin()
#3 E:\\Development\\gassner-webconnect\\vendor\\spatie\
ay\\src\\Request.php(30): Spatie\\Ray\\Payloads\\Payload->toArray()
#4 [internal function]: Spatie\\Ray\\Request->Spatie\\Ray\\{closure}()
#5 E:\\Development\\gassner-webconnect\\vendor\\spatie\
ay\\src\\Request.php(31): array_map()
#6 E:\\Development\\gassner-webconnect\\vendor\\spa at E:\\Development\\gassner-webconnect\\vendor\\spatie\
ay\\src\\Origin\\Origin.php:11)
[stacktrace]
#0 {main}
"}

**Comment by freekmurze** [comment]
Our tests indicate that everything is working like expected. I'm assuming that this is caused by something specific to your project / environment.

Could you create and share a minimal up with us that recreates the issue?

**Reply by DevRaeph** [reply]
hey there, i think i figured it out.
i'm now working on a windows 10 machine and use phpstorm direct webserver^^ 
i think switching to linux would help me out with that problem.

Thanks for your fast reply!

**Reply by sinnbeck** [reply]
Try updating to the newest version of ray. $hostname isnt required anymore
